samcli/commands/deploy/guided_context.py

```diff
@@ -285,52 +288,226 @@ def prompt_parameters(
-                        pass
-                    except NoImageFoundException as ex:
-                        raise GuidedDeployFailedError("No images found to deploy, try running sam build") from ex
-                    else:
-                        click.secho(f"\t  {image} to be pushed to {image_repositories.get(resource_id)}:{tag}")
-            click.secho(nl=True)
+        image_repositories = image_repositories.copy() if image_repositories is not None else {}
+        self.function_provider = SamFunctionProvider(stacks, ignore_code_extraction_warnings=True)
+
+        manager_helper = CompanionStackManagerHelper(
+            stack_name, region, s3_bucket, s3_prefix, self.template_file, image_repositories
+        )
+
+        create_all_repos = self.prompt_create_all_repos(manager_helper)
+        if create_all_repos:
+            image_repositories.update(manager_helper.manager.get_repository_mapping())
+        else:
+            image_repositories = self.prompt_specify_repos(manager_helper, image_repositories, self.image_repository)
+            manager_helper.update_sepcified_image_repos(image_repositories)
 
+        image_repositories = self.prompt_delete_unreferenced_repos(manager_helper, image_repositories)
+        GuidedContext.verify_images_exist_locally(self.function_provider.functions, image_repositories)
+
+        manager_helper.manager.sync_repos()
         return image_repositories
 
+    def prompt_specify_repos(
+        self,
+        manager_helper: CompanionStackManagerHelper,
+        image_repositories: Dict[str, str],
+        default_image_repo,
+    ) -> Dict[str, str]:
+        """
+        Show prompts for each function that isn't associated with a image repo
+
+        Parameters
+        ----------
+        manager_helper: CompanionStackManagerHelper
+            Instance of CompanionStackManagerHelper
+
+        image_repositories: Dict[str, str]
+            Current image repo dictionary with function logical ID as key and image repo URI as value.
+
+        default_image_repo: str
+            Default image repo URI to be shown for each function prompt.
+
+        Returns
+        -------
+        Dict[str, str]
+            Updated image repo dictionary with values(image repo URIs) filled by user input
+        """
+        image_repositories = image_repositories.copy()
+        for function_logical_id in manager_helper.missing_repo_functions:
+            image_uri = prompt(
+                f"\t {self.start_bold}ECR repository for {function_logical_id}{self.end_bold}",
+                default=default_image_repo,
+                type=click.STRING,
+            )
+            if not is_ecr_url(image_uri):
+                raise GuidedDeployFailedError(f"Invalid Image Repository ECR URI: {image_uri}")
+
+            image_repositories[function_logical_id] = image_uri
+
+        return image_repositories
+
+    def prompt_create_all_repos(self, manager_helper: CompanionStackManagerHelper) -> bool:
```

Comment:
I feel we are mixing prompt logic with computing if we need repos to be created/deleted, can we compute that outside this method and pass in the repos to be created/deleted instead? Thoughts?

Reply:
Updated prompt methods (prompt_specify_repos, prompt_create_all_repos, and prompt_delete_unreferenced_repos) to be less coupled.